Description: Fix BIO control return value on unknown operations
Author: John Baldwin <jhb@freebsd.org>
Origin: https://issues.apache.org/jira/projects/SERF/issues/SERF-198
Last-Update: 2022-03-14
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/buckets/ssl_buckets.c b/buckets/ssl_buckets.c
index b01e5359db08..3c8b7e2a685f 100644
--- a/buckets/ssl_buckets.c
+++ b/buckets/ssl_buckets.c
@@ -407,7 +407,7 @@ static int bio_bucket_destroy(BIO *bio)
 
 static long bio_bucket_ctrl(BIO *bio, int cmd, long num, void *ptr)
 {
-    long ret = 1;
+    long ret = 0;
 
     switch (cmd) {
     default:
@@ -415,6 +415,7 @@ static long bio_bucket_ctrl(BIO *bio, int cmd, long num, void *ptr)
         break;
     case BIO_CTRL_FLUSH:
         /* At this point we can't force a flush. */
+        ret = 1;
         break;
     case BIO_CTRL_PUSH:
     case BIO_CTRL_POP:
